### ISSUE TITLE
manual_backward logical bug

### ISSUE THREAD

**Issue by marsggbo** [PyTorchLightning/pytorch-lightning#8104]
Actually, this problem has been raised in issue #7698 , but it has been closed. 

This problem still exists in V1.3.7. My current solution is to comment `track_and_norm_grad` function. I would suggest adding a conditional judgment of grad clipping.

```python

    def backward(self, result, optimizer, opt_idx, *args, **kwargs):
        self.trainer.dev_debugger.track_event("backward_call")

        should_accumulate = self.should_accumulate()

        # backward can be called manually in the training loop
        if isinstance(result, torch.Tensor):
            self.trainer.accelerator.backward(result, optimizer, opt_idx, should_accumulate, *args, **kwargs)
        else:
            result.closure_loss = self.trainer.accelerator.backward(
                result.closure_loss, optimizer, opt_idx, should_accumulate, *args, **kwargs
            )

        if not self.should_accumulate() and self.grad_clip:
            # track gradients
            self.track_and_norm_grad(optimizer=optimizer)
```

> I recently also encounter this problem. I think there is indeed some logic error in `manual_loss`.
> 
> ```python
>     ....
>     self.manual_backward(loss)
> ```
> 
> * `lightning.py: manual_backward`
> 
> ```python
>     def manual_backward(self, loss: Tensor, optimizer: Optional[Optimizer] = None, *args, **kwargs) -> None:
>         ...
>         # backward
>         self._running_manual_backward = True
>         self.trainer.train_loop.backward(loss, optimizer=None, opt_idx=None, *args, **kwargs)
>         self._running_manual_backward = False
> ```
> 
> * `trainer.train_loop.backward(loss, optimizer=None,...)`
> 
> the argument `optimizer` is `None`, but the last line requires a `optimizer`
> 
> ```python
>     def backward(self, result, optimizer, opt_idx, *args, **kwargs):
>         self.trainer.dev_debugger.track_event("backward_call")
> 
>         should_accumulate = self.should_accumulate()
> 
>         # backward can be called manually in the training loop
>         if isinstance(result, torch.Tensor):
>             self.trainer.accelerator.backward(result, optimizer, opt_idx, should_accumulate, *args, **kwargs)
>         else:
>             result.closure_loss = self.trainer.accelerator.backward(
>                 result.closure_loss, optimizer, opt_idx, should_accumulate, *args, **kwargs
>             )
> 
>         if not self.should_accumulate():
>             # track gradients
>             self.track_and_norm_grad(optimizer=optimizer)   # raise error here
> ```

**Comment by carmocca** [comment]
Hi!

Setting `gradient_clip_val` or `accumulate_grad_batches` on your `Trainer` with `automatic_optimization=False` (aka using `manual_backward`) is not allowed. #7788 (the PR that closed the issue you linked) implements a better error message for that case but still stops the program.

You are not seeing this in the 1.3.7 release because the changes are in master and will be released with 1.4.0.

But in your case, you just need to remove the aforementioned flags in your Trainer definition.
